eng/common/scripts/Update-DocsMsToc.ps1

```diff
@@ -95,6 +115,116 @@ function GetPackageLookup($packageList) {
+    $mgmtTable += "[!INCLUDE [mgmt-packages]($mgmtTableLink)]`r`n"
+    Add-Content -Path $readmePath -Value $mgmtTable -NoNewline
+  }
+}
+
+# Update the metadata table on attributes: author, ms.author, ms.service
+function update-metadata-table($readmePath, $serviceName, $msService)
+{
+  $readmeContent = Get-Content -Path $readmePath -Raw
+  $null = $readmeContent -match "---`n*(?<metadata>(.*`n)*)---`n*(?<content>(.*`n)*)"
+  $metadataTable = $Matches["metadata"]
+  $metadataTable = ConvertFrom-StringData -StringData $metadataTable -Delimiter ':'
+  if (!$metadataTable) {
+    return
+  }
+  $restContent = $Matches["content"]
+  $serviceBaseName = $serviceName.ToLower().Replace(' ', '').Replace('/', '-')
+  $author = "sima-zhu" #GetPrimaryCodeOwner -TargetDirectory "/sdk/$serviceBaseName/"
+  if (!$author) {
+    LogError "Cannot fetch the author from CODEOWNER file."
+    exit 1
+  }
+  $metadataTable["author"] = $author
+  $msauthor = "sizhu" #GetMsAliasFromGithub -TenantId $tenantId -ClientId $clientId -ClientSecret $clientSecret -GithubUser $author   
+  if (!$msauthor) {
+    LogError "No ms.author found for $author. "
+    $msauthor = $author
+  }
+  $metadataTable["ms.author"] = $msauthor
+  $metadataTable["ms.service"] =  $msService
+  $metadataString = ""
+  $metadataTable.Keys | ForEach-Object {$metadataString += ($_ + ": " + $metadataTable[$_] + "`r`n")}
+  Set-Content -Path $readmePath -Value "---`n$metadataString---`n$restContent" -NoNewline
+}
+
+function generate-markdown-table($absolutePath, $readmeName, $packageInfo, $moniker) {
+  $content = "| Reference | Package | Source |`r`n|---|---|---|`r`n" 
+  # Here is the table, the versioned value will
+  foreach ($pkg in $packageInfo) {
+    if (!$pkg.VersionGA -and "latest" -eq $moniker) {
+      continue
+    }
+    if (!$pkg.VersionPreview -and "preivew" -eq $moniker) {
+      continue
+    }
+    $repositoryLink = $RepositoryUri
+    $packageLevelReame = &$GetPackageLevelReadmeFn -packageMetadata $pkg
+    $referenceLink = "$packageLevelReame-readme"
+    if (!(Test-Path "$absolutePath$referenceLink.md")) {
+      continue
+    }
+    $githubLink = $GithubUri
+    if ($pkg.PSObject.Members.Name -contains "FileMetadata") {
+      $githubLink = "$GithubUri/blob/main/$($pkg.FileMetadata.DirectoryPath)"
+    }
+    $line = "|[$($pkg.DisplayName)]($referenceLink)|[$($pkg.Package)]($repositoryLink/$($pkg.Package))|[Github]($githubLink)|`r`n"
+    $content += $line
+  }
+  Set-Content -Path "$absolutePath$readmeName" -Value $content -NoNewline
+}
+
+function generate-service-level-readme($readmeBaseName, $pathPrefix, $clientPackageInfo, $mgmtPackageInfo, $serviceName) {
+  # Add ability to override
+  # Fetch the service readme name
+  $monikers = @("latest", "preview")
+
+  $msService = GetDocsMsService -clientPackageInfo $clientPackageInfo -mgmtPackageInfo $mgmtPackageInfo -serviceName $serviceName
+  for($i=0; $i -lt $monikers.Length; $i++) {
```

Comment:
Is there a reason we use a for loop here? Best as I can tell `for($moniker in $monikers)` should get you the same result right? I don't see us use the index directly anywhere.